python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1148,23 +1186,16 @@ def my_dbt_op(dbt: DbtCliResource):
                 [assets_def]
             )
 
-            # When dbt is enabled with asset checks, we turn off any indirection with dbt selection.
-            # This way, the Dagster context completely determines what is executed in a dbt
-            # invocation.
-            if dagster_dbt_translator.settings.enable_asset_checks:
-                logger.info(
-                    "Setting environment variable `DBT_INDIRECT_SELECTION` to 'empty'. When dbt "
-                    "tests are modeled as asset checks, they are executed through explicit selection."
-                )
-                env["DBT_INDIRECT_SELECTION"] = "empty"
-
-            selection_args = get_subset_selection_for_context(
+            selection_args, indirect_selection = get_subset_selection_for_context(
                 context=context,
                 manifest=manifest,
                 select=context.op.tags.get(DAGSTER_DBT_SELECT_METADATA_KEY),
                 exclude=context.op.tags.get(DAGSTER_DBT_EXCLUDE_METADATA_KEY),
                 dagster_dbt_translator=dagster_dbt_translator,
             )
+
+            if indirect_selection:
```

Comment:
otherwise we leave it to either the default (eager), or whatever the user has set it to (e.g. cautious)